rero_ils/dojson/utils.py

```diff
@@ -178,6 +229,133 @@ def remove_trailing_punctuation(
         data.rstrip()).rstrip()
 
 
+def add_note(new_note, data):
+    """Add a new note to the data avoiding duplicate notes.
+
+    :param new_note: the note object to add
+    :type new_note: object
+    :param data: the object data on which the new note will be added
+    :type data: object
+    """
+    def note_key(note):
+        """Generate a note key by concataning the noteType and the label.
+
+        :param note: the note objet for which the key is to be generated
+        :type note: object
+        :return: the note key (concatenation of the noteType and the label)
+        :rtype: str
+        """
+        key = '|'.join((
+            note['noteType'],
+            note['label']))
+        return key
+
+    notes = data.get('note', [])
+    note_set = set()
+    for existing_note in notes:
+        note_set.add(note_key(existing_note))
+    if new_note:
+        new_note_key = note_key(new_note)
+        if new_note_key not in note_set:
+            note_set.add(new_note_key)
```

Comment:
You are right. This line was usefull when during the development, the function add_note was able to add a list of new_note. Because now the function receive only one note to add is no more needed to add the new note to the set. Good point!